consensus/mod.go

```diff
@@ -0,0 +1,64 @@
+package consensus
+
+import (
+	"github.com/golang/protobuf/proto"
+	"go.dedis.ch/fabric/crypto"
+	"go.dedis.ch/fabric/encoding"
+	"go.dedis.ch/fabric/mino"
+)
+
+// Proposal is the interface that the proposed data must implement to be
+// accepted by the consensus.
+type Proposal interface {
+	encoding.Packable
+
+	// GetHash returns the hash of the proposal.
+	GetHash() []byte
+
+	GetPublicKeys() []crypto.PublicKey
+}
+
+// Validator is the interface to implement to start a consensus.
+type Validator interface {
+	// Validate should return the proposal decoded from the message or
+	// an error if it is invalid. It should also return the previous
+	// proposal.
+	Validate(message proto.Message) (curr Proposal, prev Proposal, err error)
```

Comment:
that feels very tight to the way we make verification now, but we could imagine scenarios where a validation needs something else. What if we make the consensus use on the blockchain abstraction?
That would give something like:
```
Validate(message proto.Message) (bc Blockchain, blockIndex int, err error)
```

Reply:
My point was to build blockchain on top of the consensus but this one should not be aware of what is a blockchain. You could totally build something on top of the consensus that is unrelated to blockchain. The benefit is that the module is responsible to create the crypto links between proposals but a proposal can be anything.